lib/spack/spack/util/path.py

```diff
@@ -24,10 +24,21 @@
     'substitute_path_variables',
     'canonicalize_path']
 
+
+def get_user():
+    try:
+        # user pwd for unix systems
+        import pwd
+        return pwd.getpwuid(os.getuid()).pw_name
```

Comment:
Two things:

1. This should be outside the try/except, as it's really the import you're checking here.
2. Shouldn't this be `os.geteuid()` for the effective uid? I think this one fails with setuid, doesn't it?